packages/proxy/lib/http/response-middleware.ts

```diff
@@ -381,11 +382,82 @@ const MaybePreventCaching: ResponseMiddleware = function () {
   this.next()
 }
 
+const determineIfNeedsMultiDomainHandling = (ctx) => {
```

Comment:
Typings here might be helpful to those unfamiliar with the code, but might need some casting when the middleware actually passes in `this`, which I feel like somewhat defeats the purpose. It might be good to break out the passed in generic that includes `makeResStreamPlainText`, `incomingRes`, etc  from `ResponseMiddleware` into its own type and reference it here instead of `Http` to prevent casting issues.

```suggestion
const determineIfNeedsMultiDomainHandling = (ctx: HttpMiddlewareThis<Http>) => {
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/20450/commits/796df1d5f07e4192c7d9b5b4967b6074535e0e8b